Mathlib/Data/Polynomial/EraseLead.lean

```diff
@@ -245,6 +245,29 @@ theorem induction_with_natDegree_le (P : R[X] → Prop) (N : ℕ) (P_0 : P 0)
       exact Nat.succ_ne_zero _
 #align polynomial.induction_with_nat_degree_le Polynomial.induction_with_natDegree_le
 
+theorem coeff_mul_add_of_le_natDegree_of_eq {df dg : ℕ} {g : R[X]}
+    (hdf : natDegree f ≤ df) (hdg : natDegree g ≤ dg) :
+    (f * g).coeff (df + dg) = f.coeff df * g.coeff dg := by
+  revert hdg g
+  apply f.induction_with_natDegree_le (P_0 := by simp)
+  · intros n r _r0 hn g hdg
+    rw [mul_assoc, coeff_C_mul, coeff_C_mul, coeff_X_pow_mul', coeff_X_pow,
+      if_pos (le_add_right hn)]
+    split_ifs with H
+    · simp [H]
+    · rw [mul_zero, zero_mul]
+      apply mul_eq_zero_of_right
+      apply coeff_eq_zero_of_natDegree_lt
+      apply hdg.trans_lt
+      rw [add_comm, Nat.add_sub_assoc hn]
+      apply Nat.lt_add_of_pos_right
+      apply Nat.sub_pos_of_lt
+      exact Ne.lt_of_le' H hn
+  · intros f g _fg _gle hf hg h hle
+    rw [add_mul, coeff_add, coeff_add, add_mul]
+    congr <;> solve_by_elim
+  · assumption
```

Comment:
A proof that should fit in Mathlib.Data.Polynomial.Degree.Definitions except for requiring a lemma in your own Mathlib.Algebra.Group.UniqueProds that should probably be moved earlier:
```
  rw [coeff_mul, Finset.sum_eq_single_of_mem (df, dg)]
  · rw [Finset.Nat.mem_antidiagonal]
  rintro ⟨df', dg'⟩ hmem hne
  obtain h | hdf' := lt_or_le df df'
  · rw [coeff_eq_zero_of_natDegree_lt (hdf.trans_lt h), zero_mul]
  obtain h | hdg' := lt_or_le dg dg'
  · rw [coeff_eq_zero_of_natDegree_lt (hdg.trans_lt h), mul_zero]
  rw [Finset.Nat.mem_antidiagonal] at hmem
  have := eq_and_eq_of_le_of_le_of_add_le hdf' hdg' hmem.ge
  exact (hne.symm <| Prod.ext this.1 this.2).elim
```

Reply:
@alreadydone thank you!  I had forgotten about the `eq_and_eq_of_le_of_le_of_mul_le` lemma!

I created #6483 to relocate the lemma.  I am now going to update the current PR to depend on the relocation and to relocate `coeff_mul_add_of_le_natDegree_of_eq`.